utils/orchestrator/nomadutil.py

```diff
@@ -0,0 +1,106 @@
+# (C) Datadog, Inc. 2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import logging
+from utils.dockerutil import DockerUtil
+from utils.singleton import Singleton
+
+
+log = logging.getLogger(__name__)
+
+NOMAD_TASK_NAME = 'NOMAD_TASK_NAME'
+NOMAD_JOB_NAME = 'NOMAD_JOB_NAME'
+NOMAD_ALLOC_NAME = 'NOMAD_ALLOC_NAME'
+
+
+class NomadUtil:
+    __metaclass__ = Singleton
+
+    def __init__(self, instance=None):
+        self.docker_util = DockerUtil()
+        if instance is None:
+            log.debug("New NomadUtil instance")
+            self._container_tags_cache = {}
+            self._enabled = None
+
+    def init_platform(self, agentConfig):
+        """
+        Enable/disable Nomad support depending on the docker_orchestrator config file entry
+        :param agentConfig: dict from config.get_config()
+        """
+        self._enabled = agentConfig.get('docker_orchestrator', None) == 'nomad'
+
+    def is_enabled(self):
+        """
+        Allows user classes to check whether nomad is to be activated.
+        init_platform has to be called once first
+        :return: True/False
+        """
+        if self._enabled is None:
+            log.warning("Calling nomad.is_nomad before init_platform, replying False")
+            return False
```

Comment:
Because I specified "docker_orchestrator:nomad" in datadog.conf and not docker_daemon.yaml, which leads to this hacky way to get agentConfig.
We should probably just use a nomad_tags option in docker_daemon.yaml and move to a global orchestrator option in agent6